### ISSUE TITLE
Disallow wildcards in type alias and enum definitions

### ISSUE THREAD

**Issue by mlutze** [flix/flix#4560]
What is the use-case for this?

```scala
type alias TA = Map[_, _]

enum E {
    case C(_)
}
```

This does not seem meaningful and complicates the typing as we cannot assume the set of type variables in the alias definition or enum case is a subset of those in the declaration.

**Comment by mlutze** [comment]
No, because `TA` has no parameter. It's almost like an existential type.

**Comment by JonathanStarup** [comment]
oh cool, does `TA` have any values? how does it work? I've never seen this

**Comment by mlutze** [comment]
> oh cool, does `TA` have any values? how does it work? I've never seen this

I'm not too sure; I don't think it's well-defined so anything could happen :D

**Comment by JonathanStarup** [comment]
This seems to break the compiler in a weird way
```

type alias SF[a, b, s] = (s, (s, a) -> (s, b))

type alias ESF[a, b] = SF[a, b, _]

def test(esf: ESF[Int32, Int32]): Int32 =
    let (s0, f) = esf;
    32
```
